cypress.config.ts

```diff
@@ -1,74 +1,50 @@
+/* eslint-disable no-console */
 import { defineConfig } from 'cypress';
 import { removeDirectory } from 'cypress-delete-downloads-folder';
+import { getSpecPattern } from '@shell/utils/cypress';
 // Required for env vars to be available in cypress
 require('dotenv').config();
 
-const skipSetup = process.env.TEST_SKIP_SETUP === 'true';
-const hasCoverage = (process.env.TEST_INSTRUMENT === 'true') || false; // Add coverage if instrumented
-
 /**
- * Filter test spec paths based on env var configuration
- * @returns
+ * VARIABLES
  */
-const getSpecPattern = (): string[] => {
-  const optionalPaths = [
-    {
-      path:   'cypress/e2e/tests/setup/**/*.spec.ts',
-      active: !skipSetup
-    }
-  ];
-  const activePaths = optionalPaths.filter(({ active }) => Boolean(active)).map(({ path }) => path);
-
-  // List the test directories to be included
-  const testDirs = ['pages', 'navigation', 'global-ui'].map((dir) => `cypress/e2e/tests/${ dir }/**/*.spec.ts`);
-
-  return [
-    ...activePaths,
-    ...testDirs
-  ];
-};
+const hasCoverage = (process.env.TEST_INSTRUMENT === 'true') || false; // Add coverage if instrumented
+const testDirs = ['setup', 'pages', 'navigation', 'global-ui'];
+const skipSetup = process.env.TEST_SKIP?.includes('setup') || process.env.TEST_SKIP_SETUP === 'true'; // Keep fallback for older configurations
```

Comment:
This is used later, so it has to stay there.

```
if (!skipSetup && !process.env.CATTLE_BOOTSTRAP_PASSWORD) {
  console.log(' ❌ You must provide CATTLE_BOOTSTRAP_PASSWORD when running setup tests');
}
```

Reply:
skipSetup, which uses TEST_SKIP_SETUP, is only used in a console log rather than the functionality associated with it

Reply:
It's the log you have put before just with another name and without having the logic in place. 
Are you asking me to remove that?

Reply:
At the moment setting `process.env.TEST_SKIP_SETUP` just outputs a misleading message that setup tests will be skipped (when they won't). One of below needs to happen
- TEST_SKIP_SETUP removing from this line
- TEST_SKIP_SETUP wired in to scripts/cypress.ts as per TEST_SKIP
- TEST_SKIP removed and TEST_SKIP_SETUP used in scripts/cypress.ts

Reply:
Ok sorry but or I am missing something obvious or there's some confusion.

In the current configuration we have:

https://github.com/rancher/dashboard/blob/master/cypress.config.ts#L6C1-L6C58
```ts
const skipSetup = process.env.TEST_SKIP_SETUP === 'true';
```

https://github.com/rancher/dashboard/blob/master/cypress.config.ts#L49C1-L51C2
```ts
if (!skipSetup && !process.env.CATTLE_BOOTSTRAP_PASSWORD) {
  console.log(' ❌ You must provide CATTLE_BOOTSTRAP_PASSWORD when running setup tests'); // eslint-disable-line no-console
}
```

I have just changed the env var to this:
https://github.com/rancher/dashboard/pull/9887/files#diff-3f62e885350a48939f240b9daf40f6b06d41da7751f2b613238fc7669ac0a1b2R13
```ts
const skipSetup = process.env.TEST_SKIP?.includes('setup') || process.env.TEST_SKIP_SETUP === 'true'; // Keep fallback for older configurations
Member
```

Therefore I am not adding anything new and just edited to a new logic of the env var (keeping the old as mentioned in the comment).

Do you want to remove the log?

Also what the log is stating is correct, because we need to avoid the setup got existing instances or it would never work. Or did I miss something?

Reply:
In the old world `const skipSetup = process.env.TEST_SKIP_SETUP === 'true';` was wired in via 

```
const getSpecPattern = (): string[] => {
  const optionalPaths = [
    {
      path:   'cypress/e2e/tests/setup/**/*.spec.ts',
      active: !skipSetup
    }
  ];
```
https://github.com/rancher/dashboard/blob/master/cypress.config.ts#L17

In this PR `getSpecPattern` has been moved and now only looks at `TEST_ONLY` and `TEST_SKIP`. 

So `TEST_SKIP_SETUP` now only affects log output and not the functionality to skip setup like it used to.

Reply:
That's handled in this line
https://github.com/rancher/dashboard/pull/9887/files#diff-3f62e885350a48939f240b9daf40f6b06d41da7751f2b613238fc7669ac0a1b2R97

```ts
    specPattern:                  getSpecPattern(testDirs, process.env),
```

Reply:
Ok there's one point which I missed and is that if you use `TEST_SKIP_SETUP` this is not included anymore in the script. This means that the skip logic is using just the new pattern `TEST_SKIP=setup`.
Updated documentation and description.